File: Algorithm/Python/1000+/1839. Longest Substring Of All Vowels in Order.py
class Solution:
    def longestBeautifulSubstring(self, word: str) -> int:
        n, result = len(word), 0
        idx, left = 0, 0
        candidate = set()
        while idx < n:
            if word[idx] == 'a' and len(candidate) == 0:
                left = idx
                candidate.add(word[idx])
            valid = all(ord(word[idx]) >= ord(c) for c in candidate)
            if valid:
                candidate.add(word[idx])
                if word[idx] == 'u' and len(candidate) == 5:
                    result = max(result, idx-left+1)
                    print(idx, left)
            else:
                candidate.clear()
                continue
            idx += 1  
        return result

File: Algorithm/C++/1000+/1839. Longest Substring Of All Vowels in Order.cc
#include <string>
#include <unordered_set>

class Solution {
public:
    int longestBeautifulSubstring(string word) {
        int left, idx = 0, n = word.size();
        std::unordered_set<char> cand;
        int result = 0;
        while(idx < n) {
            if (word[idx] == 'a' && cand.size() == 0) {
                left = idx;
                cand.insert(word[idx]);
            }
            bool valid = true;
            for (auto c: cand) {
                valid &= (word[idx] >= c);
            }
            if (valid) {
                cand.insert(word[idx]);
                if (word[idx] == 'u' && cand.size() == 5) {
                    result = std::max(result, idx - left + 1);
                }
            } else {
                cand.clear();
                continue;
            }
            idx++;
        }
        return result;        
    }
};